url8db16c7d62_Arsaell_EdT_tgz-pJ8-DATA_MainJ8/src/GUI/StartFrame.java

```diff
@@ -0,0 +1,202 @@
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+
+import javax.swing.BoxLayout;
+
+public class StartFrame extends JFrame {
+
+	protected DataStore ds;
+	public FieldPanel fp;
+	public WeekPanel wp;
+	public TeacherPanel tp;
+	public ClassroomPanel cp;
+	public LinkPanel lp;
+	public GroupPanel gp;
+	public JTabbedPane tabbedPane;
+	
+	public StartFrame() {
+		super();
+
+		this.ds = new DataStore();
+		//this.ds.addFixtures();
+		
+		this.fp = new FieldPanel(this.ds);
+		this.wp = new WeekPanel(this);
+		this.tp = null;
+		this.cp = null;
+		this.gp = null;
+		this.lp = null;
+		
+		
+		this.tabbedPane = new JTabbedPane(JTabbedPane.TOP);
+		tabbedPane.addChangeListener(new ChangeListener() {
+			public void stateChanged(ChangeEvent e) {
+				//System.out.println("TabbedPane.stateChanged() : " + ds);
+				//ds.addFixtures();
+				allowLinksTab();
+				//pack();
+			}
+		});
+		
+		tabbedPane.addTab("Week", wp);
+		tabbedPane.addTab("Fields", fp);
+		tabbedPane.addTab("Teachers", tp);
+		tabbedPane.addTab("Classrooms", cp);
+		tabbedPane.addTab("Groups", gp);
+		tabbedPane.addTab("Links", lp);
+		tabbedPane.setSelectedIndex(0);
+		
+		tabbedPane.setEnabledAt(5, false);
+		
+		JButton btLoad = new JButton("Charger");
+		btLoad.setToolTipText("Charge l'exemple par défaut");
+		
+		btLoad.addActionListener(new ActionListener() {
+			
+			@Override
+			public void actionPerformed(ActionEvent arg0) {
+				ds = new DataStore();
+				ds.addFixtures();
+				
+				// On enregistre le dataStore
+				BufferedReader br;
+				try {
+					br = new BufferedReader(new FileReader(System.getProperty("user.dir") + "/dataStoreLocation.loc"));
+					String fileName = "";
+					String sCurrentLine = "";
+					while ((sCurrentLine = br.readLine()) != null) {
+						fileName = sCurrentLine;
+					}
+					File fichier =  new File(fileName) ;
+					 // ouverture d'un flux sur un fichier
+					ObjectOutputStream oos =  new ObjectOutputStream(new FileOutputStream(fichier)) ;
+
+					 // sérialization de l'objet
+					oos.writeObject(ds);
+					oos.close();
+					
+					setVisible(false);
+					new MainFrame(ds);
+				
+				} catch (IOException e1) {
+					e1.printStackTrace();
+				}
+			}
+		});
+		
+		JButton btTer = new JButton("Terminer");
+		btTer.addActionListener(new ActionListener() {
+			
+			public void actionPerformed(ActionEvent e) {
+				// On enregistre le dataStore
+				BufferedReader br;
+				try {
+					br = new BufferedReader(new FileReader(System.getProperty("user.dir") + "/dataStoreLocation.loc"));
+					String fileName = "";
+					String sCurrentLine = "";
+					while ((sCurrentLine = br.readLine()) != null) {
+						fileName = sCurrentLine;
+					}
+					File fichier =  new File(fileName) ;
+					 // ouverture d'un flux sur un fichier
+					ObjectOutputStream oos =  new ObjectOutputStream(new FileOutputStream(fichier)) ;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *oos*. The resource is referenced at lines 121, 124. The resource is closed at line 125. There are other execution paths that do not contain closure statements, for example, when ObjectOutputStream.writeObject throws an exception. Ensure *oos* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).